sdk/servicebus/azure-servicebus/migration_guide.md

```diff
@@ -1,164 +1,444 @@
+```
 
-### Sending messages
+In V7:
+```python
+with receiver:
+    received_messages = receiver.receive_messages(max_message_count=10, max_wait_time=10)
+    peeked_messages = receiver.peek_messages(max_message_count=1, sequence_number=None)
+```
 
-| In v0.50 | Equivalent in v7 | Sample |
-|---|---|---|
-| `QueueClient.from_connection_string().send()  and ServiceBusClient.from_connection_string().get_queue().get_sender().send()`| `ServiceBusClient.from_connection_string().get_queue_sender().send_messages()`| [Get a sender and send a message](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/servicebus/azure-servicebus/samples/sync_samples/send_queue.py) |
-| `queue_client.send(BatchMessage(["data 1", "data 2", ...]))`| `batch = queue_sender.create_message_batch()  batch.add_message(ServiceBusMessage("data 1"))  queue_sender.send_messages(batch)`| [Create and send a batch of messages](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/servicebus/azure-servicebus/samples/sync_samples/send_queue.py) |
+### Settling messages
 
-### Scheduling messages and cancelling scheduled messages 
+- Previously, the methods to settle messages (`complete()`, `abandon()`, `defer()` and `dead_letter()`) were on the messages themselves.
+These have been moved to the receiver in the new version, take the message as input and have the `message` suffix in their name.
+- The idea is to have the message represents just the data and not have the responsibility of any operation on the service side.
+Additionally, since a message cannot be settled if the receiver that was used to receive it is not alive, tying these operations
+to the receiver drives the message home better.
 
-| In v0.50 | Equivalent in v7 | Sample |
-|---|---|---|
-| `queue_client.get_sender().schedule(schedule_time_utc, message1, message2)` | `sb_client.get_queue_sender().schedule_messages([message1, message2], schedule_time_utc)` | [Schedule messages](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/servicebus/azure-servicebus/samples/sync_samples/schedule_messages_and_cancellation.py) |
-| `queue_client.get_sender().cancel_scheduled_messages(sequence_number1, sequence_number2)`| `sb_client.get_queue_sender().cancel_scheduled_messages([sequence_number1, sequence_number2])` | [Cancel scheduled messages](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/servicebus/azure-servicebus/samples/sync_samples/schedule_messages_and_cancellation.py)|
+In V0.50:
+```python
+
+with receiver:
+    received_message.complete()
+    # or
+    # received_message.abandon()
+    # received_message.defer()
+    # received_message.dead_letter()
+```
+
+In V7:
+```python
+with receiver:
+    receiver.complete_message(received_message)
+    # or
+    # receiver.abandon_message(received_message)
+    # receiver.defer_message(received_message)
+    # receiver.dead_letter_message(received_message)
+```
 
 ### Renewing lock on the received message
```

Comment:
Can we combine the auto lock renewal section in here? Since both topics are about lock renewals it would make sense to have a single section called `Lock renewal`

Reply:
good suggestion, I tried to combine them into one, let me know how it looks to you.